### PR TITLE
[4.0] Correcting associations icons display

### PR DESCRIPTION
**Pull request by infograf768**
The associations icons do not display correctly

### Testing Instructions
Create a 4.0 multingual site (install languages via Extension install.
Create some associations and leave some items not associated.
### Before patch
for a manager:

manager-before

for the multingual associations component

multi-before

### After patch
for a manager:

aftermanager

for the multilingual associations component:

multi_associations

Note: also corrected a css for the language switcher where inline flags were displayed vertically

### CODE REVIEW

File: administrator/components/com_associations/helpers/associations.php [hunk 1]
```diff
@@ -241,7 +241,7 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
 					$additional = '<strong>' . JText::sprintf('COM_MENUS_MENU_SPRINTF', $menutype_title) . '</strong><br />';
 				}
 
-				$labelClass  = '';
+				$labelClass  = 'badge-association';
```
Comment (C-Lodder) [inline administrator/components/com_associations/helpers/associations.php]: You can replace `badge-association` with `badge-default`?

File: administrator/templates/atum/scss/template.scss [hunk 2]
```diff
@@ -119,3 +119,20 @@
...
+.item-associations {
+  padding: 0;
+}
+
+.item-associations li {
+  display: inline-block;
+  list-style: none;
+}
+
+.badge-association {
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/template.scss]: Once the ammends above have been done, the CSS for `.badge-association` can be removed.

File: administrator/components/com_associations/helpers/associations.php (hunk 1 above)
Comment (infograf768) [inline administrator/components/com_associations/helpers/associations.php]: i did this on purpose, as we already have badge-association for the managers.

Comment (infograf768) [inline administrator/components/com_associations/helpers/associations.php]: see https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_content/helpers/html/contentadministrator.php#L76 and elsewhere

Comment (infograf768) [inline administrator/components/com_associations/helpers/associations.php]: https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_contact/helpers/html/contact.php#L75
etc.

File: administrator/templates/atum/scss/template.scss (hunk 2 above)
Comment (infograf768) [inline administrator/templates/atum/scss/template.scss]: replied above.

File: administrator/components/com_associations/helpers/associations.php (hunk 1 above)
Comment (infograf768) [inline administrator/components/com_associations/helpers/associations.php]: https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_categories/helpers/html/categoriesadministrator.php#L72

Comment (C-Lodder) [inline administrator/components/com_associations/helpers/associations.php]: Is there any reason why the `badge-association` class has been used in the first place? I don't want to reinvent the wheel here. This class uses exactly the same styling as `badge-default` in BS4, so let's just use that.

Comment (infograf768) [inline administrator/components/com_associations/helpers/associations.php]: > Is there any reason why the badge-association class has been used in the first place? 

No idea, the changes are weird and the UI is less than easy in Atum.

My thought was that someone in the future may not like the default and in this case just change the colours for badge-association. therefore not depending on BS4.
If others agree, I can modify all instances

**Comment by wilsonge** [comment]
@C-Lodder I'm going to merge this as is. Given this class is used a lot elsewhere. If you want to do a patch to remove it from everywhere and ensure the patch is correct everywhere (including associations)

**Pull request merged**